cdap-ui/app/directives/widget-container/widget-schema-editor/widget-schema-editor.js

```diff
@@ -0,0 +1,119 @@
+angular.module(PKG.name + '.commons')
+  .directive('mySchemaEditor', function($window) {
+    return {
+      restrict: 'EA',
+      scope: {
+        model: '=ngModel',
+        config: '='
+      },
+      templateUrl: 'widget-container/widget-schema-editor/widget-schema-editor.html',
+      controller: function($scope, myHelpers) {
+        $scope.options = $scope.config['schema-types'];
+        var defaultType = $scope.config['schema-default-type'];
+        console.log('test', $scope.config['schema-default-type']);
+        // Format model
+        function initialize() {
+          var obj = {};
+          if ($scope.model) {
+            obj = JSON.parse($scope.model);
```

Comment:
Can we wrap this in a ```try...catch``` block? So even if we didn't get a model we can gracefully fallback to something that is presentable.